homeassistant/components/telnet/switch.py

```diff
@@ -96,82 +97,69 @@ class TelnetSwitch(SwitchEntity):
-        command_state,
-        value_template,
-        timeout,
-    ):
+        object_id: str,
+        resource: str,
+        port: int,
+        friendly_name: str,
+        command_on: str,
+        command_off: str,
+        command_state: str | None,
+        value_template: Template | None,
+        timeout: float,
+    ) -> None:
         """Initialize the switch."""
-        self._hass = hass
         self.entity_id = ENTITY_ID_FORMAT.format(object_id)
         self._resource = resource
         self._port = port
-        self._name = friendly_name
-        self._state = False
+        self._attr_name = friendly_name
+        self._attr_is_on = False
         self._command_on = command_on
         self._command_off = command_off
         self._command_state = command_state
         self._value_template = value_template
         self._timeout = timeout
+        self._attr_should_poll = bool(command_state)
 
-    def _telnet_command(self, command):
+    def _telnet_command(self, command: str) -> str | None:
         try:
             telnet = telnetlib.Telnet(self._resource, self._port)
             telnet.write(command.encode("ASCII") + b"\r")
             response = telnet.read_until(b"\r", timeout=self._timeout)
-            _LOGGER.debug("telnet response: %s", response.decode("ASCII").strip())
-            return response.decode("ASCII").strip()
         except OSError as error:
             _LOGGER.error(
                 'Command "%s" failed with exception: %s', command, repr(error)
             )
             return None
+        _LOGGER.debug("telnet response: %s", response.decode("ASCII").strip())
+        return response.decode("ASCII").strip()
 
-    @property
-    def name(self):
-        """Return the name of the switch."""
-        return self._name
-
-    @property
-    def should_poll(self):
-        """Only poll if we have state command."""
-        return self._command_state is not None
-
-    @property
-    def is_on(self):
-        """Return true if device is on."""
-        return self._state
-
-    @property
-    def assumed_state(self):
-        """Return true if no state command is defined, false otherwise."""
-        return self._command_state is None
-
-    def update(self):
+    def update(self) -> None:
         """Update device state."""
+        if not self._command_state:
+            return
         response = self._telnet_command(self._command_state)
-        if response:
+        if response and self._value_template:
             rendered = self._value_template.render_with_possible_json_value(response)
-            self._state = rendered == "True"
         else:
             _LOGGER.warning("Empty response for command: %s", self._command_state)
+            return None
+        self._attr_is_on = rendered == "True"
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return true if no state command is defined, false otherwise."""
+        return self._command_state is None
```

Comment:
You could use the shorthand attribute for the assumed state: `_attr_assumed_state` (same as `_attr_should_poll`)